compiler/rustc_resolve/src/lib.rs

```diff
@@ -2256,29 +2256,30 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
             bug!("namespaced crate should have a parent")
         }
 
-        let namespaced_crate_names = self
+        let Some(namespaced_crate_names) = self
             .namespaced_crate_names
             .get(ident.name.as_str())
-            .map(|cnames| cnames.iter().map(|cname| cname.to_string().clone()).collect::<Vec<_>>());
-        if namespaced_crate_names.is_none() {
+            .map(|cnames| cnames.iter().map(|cname| cname.to_string().clone()).collect::<Vec<_>>())
+        else {
             bug!("shouldn't have called `create_namespaced_crate_module` for {}", ident)
         };
 
         // Make sure the namespaced crates exist
         let mut crate_nums = vec![];
-        for crate_name in namespaced_crate_names.clone().unwrap() {
-            let crate_num =
-                self.crate_loader(|c| c.maybe_process_path_extern(Symbol::intern(&crate_name)));
-            if crate_num.is_none() {
+        for crate_name in &namespaced_crate_names {
+            let Some(crate_num) =
+                self.crate_loader(|c| c.maybe_process_path_extern(Symbol::intern(&crate_name)))
+            else {
+                // FIXME: This isn't a bug, this should be reported as an error just like any other
```

Comment:
Can you explain why this isn't a bug? The reasoning was that the crate names in `namespaced_crate_names` are in the extern prelude and should be loadable. Why wouldn't they be loadable?

Reply:
I added this comment after I was trying to do `rustc --extern foo::bar=libfoo.rlib baz.rs` and hit this `bug!` macro. The reason was that I had mistakenly not made sure that `libfoo.rlib` existed first. It was a mistake I made, but something the compiler needs to be able to handle gracefully. Although perhaps the right place to handle that is somewhere else?